### PR TITLE
Code quality improvements on sagemaker operators/hook

### PR DESCRIPTION
**Pull request by vandonr-amz**
No functional change here. I'm working on some new features in those files and wanted to extract the cosmetic changes in a separate PR to ease reviewing.

 - removed an unnecessary parameter init (`self.config = config` is done in the base class ctor)
 - removed some unnecessary mocks in tests
 - tried to factor some code in tests

### CODE REVIEW

File: tests/providers/amazon/aws/hooks/test_sagemaker.py [hunk 1]
```diff
@@ -241,7 +241,7 @@
 }
 
 
-class TestSageMakerHook(unittest.TestCase):
```
Comment (vandonr-amz) [inline tests/providers/amazon/aws/hooks/test_sagemaker.py:244]: `unittest.TestCase` is incompatible with `@pytest.mark.parametrize` which I'm using below.
Given the fact that this inheritance was unnecessary, I simply removed it.

File: tests/providers/amazon/aws/hooks/test_sagemaker.py [hunk 2]
```diff
@@ -649,9 +623,8 @@ def test_training_with_logs(self, mock_describe, mock_client, mock_log_client, m
         assert mock_session.describe_training_job.call_count == 1
 
     @mock.patch.object(SageMakerHook, "get_conn")
-    def test_find_processing_job_by_name(self, mock_conn):
+    def test_find_processing_job_by_name(self, _):
```
Comment (vandonr-amz) [inline tests/providers/amazon/aws/hooks/test_sagemaker.py:625]: mock is necessary but default behavior doesn't need to be changed.

File: tests/providers/amazon/aws/operators/test_sagemaker_model.py [hunk 3]
```diff
@@ -46,40 +45,27 @@ class TestSageMakerModelOperator(unittest.TestCase):
...
+    @mock.patch.object(SageMakerHook, "describe_model", return_value="")
     @mock.patch.object(SageMakerHook, "create_model")
-    @mock.patch.object(sagemaker, "serialize", return_value="")
-    def test_integer_fields(self, serialize, mock_model, mock_client):
-        mock_model.return_value = {"ModelArn": "test_arn", "ResponseMetadata": {"HTTPStatusCode": 200}}
+    def test_execute(self, mock_create_model, _):
+        mock_create_model.return_value = {"ModelArn": "test_arn", "ResponseMetadata": {"HTTPStatusCode": 200}}
         self.sagemaker.execute(None)
+        mock_create_model.assert_called_once_with(CREATE_MODEL_PARAMS)
         assert self.sagemaker.integer_fields == EXPECTED_INTEGER_FIELDS
```
Comment (vandonr-amz) [inline tests/providers/amazon/aws/operators/test_sagemaker_model.py:54]: merging 2 tests in one

File: tests/providers/amazon/aws/hooks/test_sagemaker.py (hunk 1 above)
Comment (Taragolis) [reply tests/providers/amazon/aws/hooks/test_sagemaker.py:244]: Replace `unittest` classes by native class always welcome.

**Review by o-nikolas — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**